### PR TITLE
Always load NodeHttpClient to fix SSR

### PR DESCRIPTION
**Pull request by BrennanConroy**
https://github.com/aspnet/AspNetCore/issues/7823

### CODE REVIEW

File: src/SignalR/clients/ts/webpack.config.base.js [hunk 1]
```diff
@@ -40,7 +40,10 @@ module.exports = function (modulePath, browserBaseName, options) {
         },
         resolve: {
             extensions: [".ts", ".js"],
-            alias: options.alias,
+            alias: {
+                "./NodeHttpClient": path.resolve(__dirname, "signalr/src/empty.ts"),
```
Comment (BrennanConroy) [inline src/SignalR/clients/ts/webpack.config.base.js]: Review: This saves 3kb for the browser .js files
We could just remove this and not save the bytes if we want

File: src/SignalR/clients/ts/signalr/src/NodeHttpClient.ts [hunk 2]
```diff
@@ -1,23 +1,36 @@
...
 export class NodeHttpClient extends HttpClient {
     private readonly logger: ILogger;
-    private readonly request: Request.RequestAPI<Request.Request, Request.CoreOptions, Request.RequiredUriUrl>;
+    private readonly request: typeof requestModule;
     private readonly cookieJar: Request.CookieJar;
 
     public constructor(logger: ILogger) {
         super();
+        if (typeof requestModule === "undefined") {
+            throw new Error("error");
```
Comment (muratg) [inline src/SignalR/clients/ts/signalr/src/NodeHttpClient.ts]: Better error message? :)

**Review by muratg — APPROVED** [review]
Looks good. Will you be sending a PR patch as well? (2.2)

File: src/SignalR/clients/ts/signalr/src/NodeHttpClient.ts (hunk 2 above)
Comment (BrennanConroy) [reply src/SignalR/clients/ts/signalr/src/NodeHttpClient.ts]: Whoops

File: src/SignalR/clients/ts/webpack.config.base.js (hunk 1 above)
Comment (halter73) [reply src/SignalR/clients/ts/webpack.config.base.js]: I think this is the right amount of clever 👍

Of course, I'd like a more first-class way to do this so we wouldn't even have to to waste bytes on a not-implemented NodeHttpClient, but this is certainly better than bundling the real thing in the browser dist.

File: src/SignalR/clients/ts/signalr/src/empty.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import { HttpClient, HttpResponse } from "./HttpClient";
```
Comment (halter73) [inline src/SignalR/clients/ts/signalr/src/empty.ts]: Nit: rename to EmptyNodeHttpClient.ts or something else more descriptive.

File: src/SignalR/clients/ts/signalr/src/empty.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+import { ILogger } from "./ILogger";
+
+export class NodeHttpClient extends HttpClient {
+    // @ts-ignore
+    public constructor(logger: ILogger) {
+        super();
+    }
+
+    public send(): Promise<HttpResponse> {
+        throw new Error("Method not implemented.");
```
Comment (halter73) [inline src/SignalR/clients/ts/signalr/src/empty.ts]: This error message could be better too. It should probably recommend the correct way to consume the package from node.js.

**Comment by anurse** [comment]
@BrennanConroy reeeeeeee~~mix~~**base**

**Review by halter73 — APPROVED** [review]

**Pull request merged**